File: home/dot_local/share/lazyvim/capabilities/languages/go.mjs
import { captureCommandOutput } from "../../lib/commands.mjs";
import { defineCapability } from "../contract.mjs";

export default defineCapability({
  id: "language.go",
  requires: ["nvim"],
  enhancements: {
    nvim: [
      {
        extrasModule: "capabilities.extras.go",
        lazyvimExtras: ["lazyvim.plugins.extras.lang.go"],
        languageCases: [
          {
            language: "go",
            filename: "attachment_test.go",
            contents: "package behavior\n\nvar answer = 42\n",
            client: "gopls",
          },
        ],
      },
    ],
  },
  verify({ platform, versions }) {
    const actual = captureCommandOutput(platform.managedToolExecutable("go"), [
      "version",
    ]);
    if (!actual.startsWith(`go version go${versions.go}`)) {
      throw new Error(`Unexpected Go version: ${actual}`);
    }
  },
});

File: home/dot_local/share/lazyvim/lua/lazyvim_capabilities/capabilities/languages/go.lua
local commands = require("lazyvim_capabilities.commands")
local define = require("lazyvim_capabilities.contract")
return define({
	id = "language.go",
	requires = { "nvim" },
	enhancements = {
		nvim = {
			{
				extras_module = "capabilities.extras.go",
				lazyvim_extras = { "lazyvim.plugins.extras.lang.go" },
				language_cases = {
					{
						language = "go",
						filename = "attachment_test.go",
						contents = "package behavior\n\nvar answer = 42\n",
						client = "gopls",
					},
				},
			},
		},
	},
	verify = function(context)
		local actual = commands.capture(context.platform.tool("go"), { "version" })
		assert(vim.startswith(actual, "go version go" .. context.versions.go), "Unexpected Go version: " .. actual)
	end,
})
